SUPPORT.md

```diff
@@ -1,55 +1,19 @@
 OpenSSL User Support resources
 ==============================
 
-_Under Construction; not more than a collection of text fragments yet._
-
 See the OpenSSL website www.openssl.org for details on how to obtain
-commercial technical support. Free community support is available through the
-openssl-users email list (see
-<https://www.openssl.org/community/mailinglists.html for> further details).
-
-If you have any problems with OpenSSL then please take the following steps
-first:
-
- - Download the latest version from the repository
-   to see if the problem has already been addressed
- - Configure with no-asm
- - Remove compiler optimization flags
-
-If you wish to report a bug then please include the following information
-and create an issue on GitHub:
-
- - OpenSSL version: output of 'openssl version -a'
- - Configuration data: output of 'perl configdata.pm ==dump'
- - OS Name, Version, Hardware platform
- - Compiler Details (name, version)
- - Application Details (name, version)
- - Problem Description (steps that will reproduce the problem, if known)
- - Stack Traceback (if the application dumps core)
-
-Just because something doesn't work the way you expect does not mean it
-is necessarily a bug in OpenSSL. Use the openssl-users email list for this type
-of query.
-
-For *questions* on how to use OpenSSL or what went wrong when you
-tried something, our primary resource is the mailing list
-openssl-users@openssl.org, where you can get help from others in the
-OpenSSL community (which includes the developers as time permits).
-
-Only subscribers can post to openssl-users@openssl.org (although the
-archives are public).
-For more information, see <https://www.openssl.org/community/mailinglists.html>
+commercial technical support.
 
-You have general questions about using OpenSSL
-----------------------------------------------
+If you have general questions about using OpenSSL
+-------------------------------------------------
 
 In this case the [openssl-users] mailing list is the right place for you.
 The list is not only watched by the OpenSSL team members, but also by many
 other OpenSSL users. Here you will most likely get the answer to your questions.
 An overview over the [mailing lists](#mailing-lists) can be found below.
 
-You found a Bug
----------------
+If you think you found a Bug
+----------------------------
 
 If you have any problems with OpenSSL then please take the following steps first:
```

Comment:
The implication of this section is "do these things first and if the problem goes away then you're done - no further action necessary". Whereas in fact there are different actions required depending on what happens. If the first two bullets solve the issue - then really nothing further does need to be done. If the 3rd one works then actually this indicates a bug in OpenSSL which we'd like to hear about - the user might be tempted to use this workaround as a "solution" to their problem - but we really should not be encouraging that. If the fourth bullet works then this most likely indicates a compiler bug - the user might consider raising a bug against the compiler (incidentally there is no clue given as to *how* a user should remove compiler optimizations)

We might want to consider breaking this list up a bit.

I'd still like more caveats on the no-asm suggestion below. It is a useful diagnostic test to figure out what kind of bug this is. But it doesn't mean the problem is "solved". Possibly a smple statement saying we don't recommend no-asm builds for production would be sufficient.

I'm actually not too sure why we give "no-asm" such prominence in this document. Whether the problem goes away or not with no-asm builds is a useful piece of information in an issue report - but its not top of the list of most important things to know.

Possibly it should be moved to the section "Open an Issue" as a piece of information we would want to know. Or maybe even remove it altogether.

Reply:
> The implication of this section is "do these things first and if the problem goes away then you're done - no further action necessary".

Hmm, that's not how I read it, especially since this section contains the action sections "Open an Issue" and "Submit a Pull Request".

Reply:
I added a small NOTE at the beginning of this section to clarify the intent.  Does that answer your concerns well enough?

Reply:
> I'm actually not too sure why we give "no-asm" such prominence in this document.

The (no-asm) C source is our reference code that's supposed to work under most if not any condition.  The asm code has had history of bugs, and considering how well versed many are with assembler these days, that are hard to figure out.  Trying with no-asm is a clear way to figure out if the issue is in the assembler code or not.

Reply:
Essentially, this section has *never* been about getting your production installation to work.